### ISSUE TITLE
Unknown Error

### ISSUE THREAD

**Issue by JahangirButt-BU** [minar09/cp-vton-plus#29]
got this error with the line: python train.py --name GMM --stage GMM --workers 4 --save_count 5000 --shuffle
![image](https://user-images.githubusercontent.com/72991763/96287274-8a9d2e80-0f96-11eb-8c86-e826a0691c8e.png)
![image](https://user-images.githubusercontent.com/72991763/96287308-97ba1d80-0f96-11eb-8b34-3749109335ca.png)

and got this with: python train.py --name TOM --stage TOM --workers 4 --save_count 5000 --shuffle
![image](https://user-images.githubusercontent.com/72991763/96287200-68a3ac00-0f96-11eb-8542-7f17ee6cc472.png)

can you guide me, on how to solve these error
downloaded the data, model and did according to the read me

![image](https://user-images.githubusercontent.com/72991763/96287340-a3a5df80-0f96-11eb-8b14-5abe388d0c39.png)

![image](https://user-images.githubusercontent.com/72991763/96287358-aa345700-0f96-11eb-9ae1-3c142f5b4dd4.png)

**Comment by minar09** [comment]
Seems like CUDA version problem in your machine, you can try upgrading maybe.

**Comment by JahangirButt-BU** [comment]
WHT is a Cuda?

**Comment by JahangirButt-BU** [comment]
and is NVidia GPU a must or do I compulsory need NVidia or is any other GPU fine?
can I run this without a GPU?

**Comment by minar09** [comment]
Yes, you can run with CPU only. If you want to run in GPUs, mention the gpu ids in the run arguments of train.py/test.py
Any other GPU should be okay as long as NVIDIA driver is available for that GPU I guess. I could be wrong. Please try to find out online in details.

**Comment by JahangirButt-BU** [comment]
![image](https://user-images.githubusercontent.com/72991763/96336303-a276c000-1033-11eb-90fe-2dadde615417.png)
![image](https://user-images.githubusercontent.com/72991763/96337276-bc67d100-103a-11eb-91f3-f0120072db35.png)
![image](https://user-images.githubusercontent.com/72991763/96337281-c8ec2980-103a-11eb-9b4b-a09b5c6f6632.png)

cuda error is resolved by download pytorch again, now
i have only CPU and this error showing

**Comment by minar09** [comment]
You can modify the code to run without GPU.
For example as shown below, make `use_CUDA=False`
![image](https://user-images.githubusercontent.com/3684134/96337347-77d34880-10c1-11eb-858a-f1633209cc79.png)

**Comment by JahangirButt-BU** [comment]
oh thankyou so much, greatly appreciated. ill close this if issue when successfully run. now  facing issue number #6

**Comment by JahangirButt-BU** [comment]
![image](https://user-images.githubusercontent.com/72991763/96347741-8cc7c180-1058-11eb-99c8-15668c3fbb8a.png)
![image](https://user-images.githubusercontent.com/72991763/96347749-95b89300-1058-11eb-9a2e-926aeb7ffbee.png)

did as u said, did cuda=False in all locations ( inside network file) still getting this

**Comment by JahangirButt-BU** [comment]
resolved the above according to issue #21 but still facing this 
![image](https://user-images.githubusercontent.com/72991763/96348311-a965f880-105c-11eb-842d-bc5b41050f51.png)
![image](https://user-images.githubusercontent.com/72991763/96348313-b4208d80-105c-11eb-9170-7c94a2877eee.png)

**Comment by JahangirButt-BU** [comment]
![image](https://user-images.githubusercontent.com/72991763/96348529-63aa2f80-105e-11eb-84de-6cade23094dc.png)

is this how, its suppose to work?

**Comment by minar09** [comment]
@JahangirButt-BU , yes its training now. If you have downloaded the pre-trained model, you can do testing directly as well. Good luck.

**Comment by JahangirButt-BU** [comment]
thankyou for all the help, training and testing going good.

but now I want to put a specific cloth onto a specific person, i understand these points:
1. image (image of a person, crop/resize to 192 x 256 (width x height) pixels)
2. image-parse (you can generate with LIP_JPPNet or CIHP_PGN pretrained networks from the person image. If you use LIP_JPPNet/LIP_SSL, then you need to run dataset_neck_skin_connection.py and body_binary_masking.py after generating the segmentation files. For CIHP_PGN segmentation, see this comment)
3. cloth (in-shop cloth image, crop/resize to 192 x 256 (width x height) pixels)
4. cloth-mask (binary mask of cloth image, you can generate it with simple pillow/opencv function)
5. pose (pose keypoints of the person, generate with openpose COCO-18 model )

but where do I put my specified image and cloth image or should some change be made in code?

**Comment by JahangirButt-BU** [comment]
is there any way to download this file

![image](https://user-images.githubusercontent.com/72991763/96377125-23b77b00-1138-11eb-80d9-7abdae1236fc.png)

cuz the training is taking long

**Comment by minar09** [comment]
Please check the VITON dataset, test folder for custom images. You can arrange following exactly as their folders.
Also, you need to make a test_pairs.txt file for your custom images. Follow the VITON dataset format to keep same arrangements, otherwise you can modify the code.

**Comment by JahangirButt-BU** [comment]
![image](https://user-images.githubusercontent.com/72991763/96586712-3778f380-1296-11eb-9c2a-d9c4d754c8a7.png)


Any Clue on how to fix this error. I generated image-parse and image-parse-new using LIP_JPPNet then running ' dataset_neck_skin_correction.py ' i get this

**Comment by minar09** [comment]
You are probably adding RGB segmentation files, instead of using the grayscale files e.g. [0,20]. Please check the viton dataset segmentation format.

**Comment by JahangirButt-BU** [comment]
![image](https://user-images.githubusercontent.com/72991763/96591476-2501b880-129c-11eb-887d-467585ee4deb.png)
![image](https://user-images.githubusercontent.com/72991763/96591512-2af79980-129c-11eb-868d-3c5b0f3d2a1d.png)
![image](https://user-images.githubusercontent.com/72991763/96591531-321ea780-129c-11eb-8744-a2a00549203c.png)
![image](https://user-images.githubusercontent.com/72991763/96591573-3c40a600-129c-11eb-9d43-beb956d202c5.png)

yes i arranged it according to the viton dataset segmentation format

**Comment by JahangirButt-BU** [comment]
can you also provide the code for generating cloth mask i am unable to find the right code/function to generate the desiired mask of cloths

**Comment by minar09** [comment]
> can you also provide the code for generating cloth mask i am unable to find the right code/function to generate the desiired mask of cloths

Just use the opencv rgb2gray function or something as you see fit for your images.

**Comment by JahangirButt-BU** [comment]
thankyou, any clue on the other issue?
![Uploading image.png…]()
